### PR TITLE
S3/sops secrets fix for optional configs

### PR DESCRIPTION
**Pull request by philipcristiano**
Based on #102 

OPTIONAL_CONFIGS: Allow missing files unless deploying

This fixes a case that prevented commands from running when using S3/sops secrets without the use of ROK8S_ENABLE_OPTIONAL_CONFIGS.

The pattern for s3/sops secrets is to define a file in `S3_SECRETS` or `SOPS_SECRETS` as well as in `SECRETS`. The likely case with this setup is that the `SECRETS` file wouldn't be created until after running `k8s-secrets-from-s3` or `k8s-sops-secret-decrypt`. This aren't runnable though because the config reading can't find the file in `SECRETS`.

This change makes it so that the `ROK8S_ENABLE_OPTIONAL_CONFIGS` is only used for the deploy commands, allowing other commands to run even if files are missing (the previous default).

### CODE REVIEW

File: bin/k8s-read-config [hunk 1]
```diff
@@ -33,6 +33,13 @@ else
   exit 1
 fi
 
+if [[ -n ${ROK8S_DEPLOY} ]] && [[ -z ${ROK8S_ENABLE_OPTIONAL_CONFIGS} ]]; then
+    allow_missing_files=false
+else
+    allow_missing_files=true
+fi
+echo missing files ${allow_missing_files} ${ROK8S_DEPLOY} ${ROK8S_ENABLE_OPTIONAL_CONFIGS}
```
Comment (coreypobrien) [inline bin/k8s-read-config]: Nit: we should probably delete this debugging (or alternatively make it clearer so it is less debug-y)

Comment (philipcristiano) [reply bin/k8s-read-config]: Removed!

**Comment by philipcristiano** [comment]
We should check for sops files normally and likely have a similar behavior with optional config flag. Secrets are a bit messy in that regard at the moment where the current usage of declaring the file in `SECRETS` as well means that it will always have to be optional. The sops bit is a bit messy as it also doesn't fail if there ciphertext files are missing :/ 

For this I'd like to at least get back to a place where secrets worked as they did before. I was thinking that we might eventually add a list of fields which need to be checked (to allow checking only the fields that would be used like in the case for sops).

**Comment by philipcristiano** [comment]
@coreypobrien thoughts on fix vs future work? I'd like to see sops/s3 secrets useful again even if not ideal.

**Pull request merged**